CodeGeneration/Sources/SyntaxSupport/RawSyntaxNodeKind.swift

```diff
@@ -0,0 +1,47 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+/// The raw counterpart of ``SyntaxNodeKind``.
+struct RawSyntaxNodeKind: IdentifierConvertible, TypeConvertible {
+  var syntaxNodeKind: SyntaxNodeKind
+
+  var identifier: TokenSyntax {
+    self.syntaxNodeKind.identifier
+  }
+
+  var isBase: Bool {
+    self.syntaxNodeKind.isBase
+  }
+
+  var syntaxType: TypeSyntax {
+    "Raw\(self.syntaxNodeKind.syntaxType)"
+  }
+
+  var protocolType: TypeSyntax {
+    switch self {
+    case .syntax, .syntaxCollection:
+      return "RawSyntaxNodeProtocol"
+    default:
+      return "\(self.syntaxType)NodeProtocol"
+    }
+  }
+
+  var isAvailableInDocc: Bool {
```

Comment:
Are any raw nodes available in docc? They are all marked as `@_spi(RawSyntax)`, so I would expect that none of them are available in docc.

Reply:
I didn't know that, I'll just set it to false then.